src/main/java/de/dennisguse/opentracks/content/provider/CustomContentProvider.java

```diff
@@ -356,11 +362,46 @@ private Uri insertMarker(Uri url, ContentValues contentValues) {
         throw new SQLException("Failed to insert a marker " + url);
     }
 
+    /**
+     * Return the string representing the query that compute sensor stats from trackpoints table.
+     * It computes time weighted heart rate, cadence and power averages. So the more time it takes the more important is.
+     * Also, it computes heart rate and cadence maximum.
+     * Finally, it ignores pause time for whatever reason (SEGMENT_START_MANUAL, SEGMENT_START_AUTOMATIC).
```

Comment:
I would argue that a START_AUTOMATIC should be considered as it only denotes a gap in the GPS track (due to fast movement, low reception etc).
Example: 
If I use a cycling sensor (distance/speed), this sensor data is stored for all trackpoints of type TRACKPOINT, START_AUTOMATIC, and (I guess even) END_MANUAL. 
If I am cycling in the mountains, GPS reception is likely to be varying a lot and this will likely result in START_AUTOMATIC (like going downhill with a racing bike at 60km/h).
And in my opinion, these should be part of the AVG/MAX computation.

In data:
t=0, START_MANUAL
t=10, TRACKPOINT, speed=5
t=20, TRACKPOINT, speed=10
t=25, START_AUTOMATIC speed=60
t=27, START_AUTOMATIC speed=60
t=30, TRACKPOINT, speed=15
t=30, END_MANUAL, speed=15

I was thinking about this while working on the "distance from sensor" task and for the overall sum to match, we need to consider all trackpoints that have distance (relative to previous).
So, basically everything except START_MANUAL, which should not have any sensor data (as no connection to sensor is yet established) and it has not duration to previous trackpoint.

Reply:
Umm, okay, I understand. I was thinking on those moments when you take a rest in your route and forget pause the activity. I guess we cannot do anything here.

So, yes, you are right.